designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputYesNoField.kt

```diff
@@ -0,0 +1,97 @@
+ * @param modifier allows a modifier to be passed externally
+ * @param state Manages the InputShell state
+ * @param supportingText is a list of SupportingTextData that
+ * manages all the messages to be shown
+ * @param legendData manages the legendComponent
+ * @param isRequired controls whether the field is mandatory or not
+ * @param itemSelected controls which item is selected.
+ * @param onItemChange is a callback to notify which item has changed into the block.
+ */
+@Composable
+fun InputYesNoField(
+    title: String,
+    modifier: Modifier = Modifier,
+    state: InputShellState = InputShellState.UNFOCUSED,
+    supportingText: List<SupportingTextData>? = null,
+    legendData: LegendData? = null,
+    isRequired: Boolean = false,
+    itemSelected: InputYesNoFieldValues? = null,
+    onItemChange: (InputYesNoFieldValues?) -> Unit,
+) {
+    InputShell(
+        modifier = modifier.testTag("INPUT_YES_NO_FIELD"),
+        isRequiredField = isRequired,
+        title = title,
+        state = state,
+        legend = {
+            legendData?.let {
+                Legend(legendData, modifier.testTag("INPUT_YES_NO_FIELD_LEGEND"))
+            }
+        },
+        supportingText = {
+            supportingText?.forEach { label ->
+                SupportingText(
+                    label.text,
+                    label.state,
+                    modifier = modifier.testTag("INPUT_YES_NO_FIELD_SUPPORTING_TEXT"),
+                )
+            }
+        },
+        inputField = {
+            val options = InputYesNoFieldValues.values().map {
+                RadioButtonData(
+                    it.value,
+                    itemSelected == it,
+                    state != InputShellState.DISABLED,
+                    provideStringResource(it.value.lowercase(Locale.getDefault())),
+                )
+            }
+            RadioButtonBlock(
+                Orientation.HORIZONTAL,
+                options,
+                options.find { it.selected },
+            ) { radioButtonData ->
+                onItemChange.invoke(
+                    InputYesNoFieldValues.values().firstOrNull { it.name.equals(radioButtonData.uid, true) },
+                )
+            }
+        },
+        primaryButton = {
+            val isClearButtonVisible = itemSelected != null && state != InputShellState.DISABLED
+            if (isClearButtonVisible) {
+                IconButton(
+                    modifier = Modifier.testTag("INPUT_YES_NO_FIELD_CLEAR_BUTTON"),
+                    icon = {
+                        Icon(
+                            imageVector = Icons.Outlined.Cancel,
+                            contentDescription = "Icon Button",
+                        )
+                    },
+                    onClick = {
+                        onItemChange.invoke(null)
+                    },
+                )
+            }
+        },
+    )
+}
+
+enum class InputYesNoFieldValues(val value: String) {
```

Comment:
this values should be taken from string resources in order to make translations in the future.

Reply:
@andresmr The value already uses string resources for the translations.
` provideStringResource(it.value.lowercase(Locale.getDefault()))`